scripts/statistics.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+const bcd = require('..');
+
+const browsers = ['chrome', 'chrome_android', 'edge', 'firefox', 'ie', 'safari', 'safari_ios', 'webview_android'];
+let stats = { total: { all: 0, true: 0, null: 0, real: 0 } };
+browsers.forEach(browser => {
+  stats[browser] = { all: 0, true: 0, null: 0, real: 0 }
+});
+
+const checkSupport = (supportData, type) => {
+ if (Array.isArray(supportData)) {
+  supportData.forEach(item => {
+    if (item.version_added === type || item.version_removed === type) {
```

Comment:
If I'm reading this correctly (a questionable assumption), then this seems to return on the first instance of an array matching `type` (`true` or `null`, depending on the caller), rather than considering every entry in the array.

So something like this:

```json
[
  {
    "version_added": null,
  },
  {
    "prefix": "-webkit-",
    "version_added": null,
    "version_removed": true,
  }
]
```

Will count toward `true` once, and `null` once (not `true` once and `null` twice). Is that intended? I think this speaks to the semantics of the count: do you want to count "the number of `null`s and `true`s in the data set"? Or do you intend to count "the number of browser entries that contain one or more `null`s and one or more `true`s in the data set"?

If it's the latter, I think there's some rewriting to do. If it's the former, then I think something like 

```
      return (item.version_added === type || item.version_removed === type);
```

in place of `if (…) return true` is a bit more direct

Reply:
Thanks, another good thought! Data is hard :)
So, if I see it correctly, the above case happens 6 times for `null` and 420 times for `true`. So, it matters which semantics we're choosing here. I think we should go with "the number of browser entries that contain one or more nulls and one or more trues in the data set". In other words give an array support statement the same weight as a single support statement. I hope that my updated code does that now.
I think the more direct statement shouldn't be placed in the array support iteration now, as it would always set it to the last checked item. I've used it for the single support check, though.
Does all this make sense?

Reply:
Yes, this works for me both on the code and the semantics of the count. Thank you Florian!